compiler/ilgen/IlBuilder.cpp

```diff
@@ -932,24 +932,48 @@ IlBuilder::ConstInteger(TR::IlType *intType, int64_t value)
 TR::IlValue *
 IlBuilder::ConvertTo(TR::IlType *t, TR::IlValue *v)
    {
-   appendBlock();
    TR::DataType t1 = v->getSymbol()->getDataType();
    TR::DataType t2 = t->getPrimitiveType();
    if (t1 == t2)
       {
       TraceIL("IlBuilder[ %p ]::%d is ConvertTo (already has type %s) %d\n", this, v->getCPIndex(), t->getName(), v->getCPIndex());
       return v;
       }
+   TR::IlValue *convertedValue = convertTo(t, v, false);
+   TraceIL("IlBuilder[ %p ]::%d is ConvertTo(%s) %d\n", this, convertedValue->getCPIndex(), t->getName(), v->getCPIndex());
+   ILB_REPLAY("%s = %s->ConvertTo(%s, %s);", REPLAY_VALUE(convertedValue), REPLAY_BUILDER(this), REPLAY_TYPE(t), REPLAY_VALUE(value));
+   return convertedValue;
+   }
 
-   TR::ILOpCodes convertOp = TR::DataType::getDataTypeConversion(v->getSymbol()->getDataType(), t->getPrimitiveType());
+TR::IlValue *
+IlBuilder::UnsignedConvertTo(TR::IlType *t, TR::IlValue *v)
+   {
+   TR::DataType t1 = v->getSymbol()->getDataType();
+   TR::DataType t2 = t->getPrimitiveType();
+   if (t1 == t2)
+      {
+      TraceIL("IlBuilder[ %p ]::%d is UnsignedConvertTo (already has type %s) %d\n", this, v->getCPIndex(), t->getName(), v->getCPIndex());
+      return v;
+      }
+   TR::IlValue *convertedValue = convertTo(t, v, true);
+   TraceIL("IlBuilder[ %p ]::%d is UnsignedConvertTo(%s) %d\n", this, convertedValue->getCPIndex(), t->getName(), v->getCPIndex());
+   ILB_REPLAY("%s = %s->UnsignedConvertTo(%s, %s);", REPLAY_VALUE(convertedValue), REPLAY_BUILDER(this), REPLAY_TYPE(t), REPLAY_VALUE(value));
+   return convertedValue;
+   }
+
+TR::IlValue *
+IlBuilder::convertTo(TR::IlType *t, TR::IlValue *v, bool needUnsigned)
+   {
+   TR::DataType t1 = v->getSymbol()->getDataType();
+   TR::DataType t2 = t->getPrimitiveType();
+
+   appendBlock();
+   TR::ILOpCodes convertOp = ILOpCode::getProperConversion(v->getSymbol()->getDataType(), t->getPrimitiveType(), needUnsigned);
```

Comment:
Should this be 

```c++
TR::ILOpCodes convertOp = ILOpCode::getProperConversion(t1, t2, needUnsigned);
```

Otherwise, why have the

```c++
TR::DataType t1 = v->getSymbol()->getDataType();
TR::DataType t2 = t->getPrimitiveType();
```